x/erc20/keeper/ibc_callbacks.go

```diff
@@ -62,11 +62,8 @@ func (k Keeper) OnRecvPacket(
 	}
 	recipient := sdk.AccAddress(recipientBz)
 
-	senderBz, err := k.addrCodec.StringToBytes(data.Sender)
-	if err != nil {
-		return channeltypes.NewErrorAcknowledgement(errorsmod.Wrap(err, "invalid sender"))
-	}
-	sender := sdk.AccAddress(senderBz)
+	// decides if sending to self without decoding sender
+	sendToSelf := data.Sender == data.Receiver
```

Comment:
The use of this doesn't make sense to me and we should actually remove this case from mint.go